### PR TITLE
Rename `Spend` to `SpendConditions`

### PR DESCRIPTION
**Pull request by matt-o-how**
### Purpose:
Rename `Spend` to a less ambiguous and more accurate name: `SpendConditions`,

When chia_rs bumps, and changes name as well, we will need to update the import in `spend_bundle_conditions.py`

### CODE REVIEW

File: chia/types/spend_bundle_conditions.py [hunk 1]
```diff
@@ -3,5 +3,9 @@
 import chia_rs
 
 ELIGIBLE_FOR_DEDUP = chia_rs.ELIGIBLE_FOR_DEDUP
-Spend = chia_rs.Spend
+if hasattr(chia_rs, "SpendConditions"):
+    SpendConditions = chia_rs.SpendConditions
+else:
+    # Fallback to the old name if the new name does not exist
+    SpendConditions = chia_rs.Spend
```
Comment (arvidn) [inline chia/types/spend_bundle_conditions.py:10]: I would hold off on this and make this change when we bump `chia_rs` next

**Review by arvidn — CHANGES_REQUESTED** [review]

File: chia/types/spend_bundle_conditions.py (hunk 1 above)
Comment (arvidn) [reply chia/types/spend_bundle_conditions.py:10]: I'm referring to supporting both old and new `chia_rs`. I think it's fine to only support the version that we depend on right now.

Comment (arvidn) [reply chia/types/spend_bundle_conditions.py:10]: the circular dependency between chia_rs and chia-blockchain makes this a bit tricky

**Review by arvidn — DISMISSED** [review]

**Comment by Starttoaster** [comment]
@matt-o-how @arvidn - This PR needs a determination on the coverage diff prior to merge

**Review by arvidn — APPROVED** [review]

**Pull request merged**